syncd/ComparisonLogic.cpp

```diff
@@ -2836,6 +2838,21 @@ void ComparisonLogic::cretePreMatchForAclEntries(
                 continue;
             }
 
+            if (cPrio->getStrAttrValue() == PFC_ACL_RULE_PRIORITY)
+            {
+                // this is special case, all pfc rules to prevent packet storm
+                // will have same priority, we will need to check SAI_ACL_ENTRY_ATTR_FIELD_TC
+
+                auto cTc = cAclEntry->getSaiAttr(SAI_ACL_ENTRY_ATTR_FIELD_TC);
+                auto tTc = tAclEntry->getSaiAttr(SAI_ACL_ENTRY_ATTR_FIELD_TC);
+
+                if (!cTc || !cTc || cTc->getStrAttrValue() != tTc->getStrAttrValue())
+                {
+                    // TC attribute is different, not looking for this one
+                    continue;
+                }
+            }
+
             // at this point current and temporary acl entry share the same
             // priority, then we can assume that those objects are the same
```

Comment:
I think this comment needs to be updated? Now not just priority, but also TC field may be same.

Reply:
updated